web/src/containers/activity/EntryDetails.js

```diff
@@ -67,41 +70,35 @@ const EntryDetails = ({ activity, aKey, num }) => {
   );
 
   return (
-    <div
-      id={`activity-${aKey}`}
-      className={`activity--body activity--body__${
-        collapsed ? 'collapsed' : 'expanded'
-      } activity--body__${num === 1 ? 'first' : 'notfirst'}`}
-      ref={container}
-    >
-      <Review heading={title} headingLevel={4} editContent={editContent} />
-      <div className={collapsed ? 'visibility--print' : ''}>
-        <Overview aKey={aKey} />
-        <Goals aKey={aKey} />
-        <Schedule aKey={aKey} />
-        <Costs aKey={aKey} />
-        <ContractorResources aKey={aKey} />
-        <CostAllocate aKey={aKey} />
-        <StandardsAndConditions aKey={aKey} />
-        <Button variation="primary" onClick={() => setCollapsed(true)}>
-          Done
-        </Button>
+    <Provider value={{ activity, index }}>
```

Comment:
This is a context provider. Any context consumers _down_ the tree from here will be able to access these values. I'm stashing off the activity and index. Now we don't have to pass anything down - child components can just grab the activity from the context.

One thing we should dig into is re-renders. I _think_ a component will re-render any time its context changes. We don't really want that - e.g., if an activity name changes, we don't need to activity goals to re-render. So maybe we should only put the index in the context and use that to go fetch the activity. 🤔